resolver.go

```diff
@@ -432,7 +440,7 @@ func (r *Resolver) resolveCNAMEs(ctx context.Context, qname, qtype string, crrs
 		logCNAME(crr.String(), depth)
 		crrs, _ := r.resolve(ctx, crr.Value, qtype, depth)
 		for _, rr := range crrs {
-			r.cache.add(qname, rr)
+			r.cache.add(rr.Name, rr)
```

Comment:
This doesn't look right. Why this change?

Reply:
Without it, the `TestGoogleCNAME` `r.ResolveErr("translate.google.com", "A")` was returning the `A` record of the NS because the cache looked like that:

Cache entry:
```golang
[...]
"translate.google.com.":
            dnsr.entry{
                dnsr.RR{Name:"l.google.com.", Type:"NS", Value:"ns1.google.com.", TTL:0, Expiry:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}:struct {}{},
                dnsr.RR{Name:"l.google.com.", Type:"NS", Value:"ns2.google.com.", TTL:0, Expiry:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}:struct {}{},
                dnsr.RR{Name:"l.google.com.", Type:"NS", Value:"ns3.google.com.", TTL:0, Expiry:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}:struct {}{},
                dnsr.RR{Name:"l.google.com.", Type:"NS", Value:"ns4.google.com.", TTL:0, Expiry:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}:struct {}{},
                dnsr.RR{Name:"ns1.google.com.", Type:"A", Value:"216.239.32.10", TTL:0, Expiry:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}:struct {}{},
                dnsr.RR{Name:"ns2.google.com.", Type:"A", Value:"216.239.34.10", TTL:0, Expiry:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}:struct {}{},
                dnsr.RR{Name:"ns3.google.com.", Type:"A", Value:"216.239.36.10", TTL:0, Expiry:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}:struct {}{},
                dnsr.RR{Name:"ns4.google.com.", Type:"A", Value:"216.239.38.10", TTL:0, Expiry:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}:struct {}{},
                dnsr.RR{Name:"translate.google.com.", Type:"CNAME", Value:"www3.l.google.com.", TTL:0, Expiry:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}:struct {}{}
}```

And since there are 'A' records in the cache, we get the IP of the NS in the return:
```golang
dnsr.RRs{dnsr.RR{Name:"translate.google.com.", Type:"CNAME", Value:"www3.l.google.com.", TTL:0, Expiry:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC)}, dnsr.RR{Name:"www3.l.google.com.", Type:"A", Value:"142.250.179.206", TTL:0, Expiry:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC)}}
```

With my modification, the cache only contains:
```golang
"translate.google.com.":
    dnsr.entry{dnsr.RR{Name:"translate.google.com.", Type:"CNAME", Value:"www3.l.google.com.", TTL:0, Expiry:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}:struct {}{}}
```

Another way to maybe fix the issue would be to check `rr.Name == qname` [in cacheGet](https://github.com/domainr/dnsr/blob/main/resolver.go#L483).

Reply:
I can push this second option if you want.